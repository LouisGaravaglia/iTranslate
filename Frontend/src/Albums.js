import React from 'react';
import {Spring} from 'react-spring/renderprops';
import './App.css';
//COMPONENT IMPORTS
import SearchResultList from "./SearchResultList";
//REDUX IMPORTS
import {useDispatch, useSelector} from "react-redux";
import {getTracks} from "./actionCreators/BrowseRoute/Artists/getTracksCreator";
import {resetStore} from "./actionCreators/resetStoreCreator";

function Albums({typeOfAlbums}) {
  //REDUX STORE
  const dispatch = useDispatch();
  const albums = useSelector(store => store.albums);

////////////////////////////////////////////////////  HANDLE CLICK FUNCTIONS  ////////////////////////////////////////////////////

  const handleAlbumClick = async (albumId) => {
    dispatch(getTracks(albumId));
    dispatch(resetStore("lyrics", "translation"));
  };

////////////////////////////////////////////////////  RETURN  ////////////////////////////////////////////////////

  return (
    <Spring
      from={{opacity: 0}}
      to={{opacity: 1}}
      config={{delay: 300, duration: 300}}
    >
      {props => (
        <div style={props}>

          <div className="Main-Container">
            <SearchResultList key={albums[0].albumId} typeOfResults="albums" resultsArray={albums} handleSearch={handleAlbumClick} itemsPerPage={3} typeOfAlbums={typeOfAlbums}/>
          </div>

        </div>
      )}
    </Spring>
  );
};

export default Albums;