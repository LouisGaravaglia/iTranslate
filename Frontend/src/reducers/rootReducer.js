import {combineReducers} from "redux";
import translationReducer from "./translationReducer";
import errorsReducer from "./errorsReducer";
import lyricsReducer from "./lyricsReducer";
import resultsReducer from "./resultsReducer";

const rootReducer = combineReducers({results: resultsReducer, translation: translationReducer, lyrics: lyricsReducer, errors: errorsReducer});

export default rootReducer;