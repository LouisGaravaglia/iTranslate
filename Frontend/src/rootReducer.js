// const SEED_ARTISTS = [{name: "J Balvin", id:"1vyhD5VmyZ7KMfW5gqLgo5"}, {name: "Bad Bunny", id:"4q3ewBCX7sLwd24euuV69X"}, {name: "Maluma", id:"1r4hJ1h58CWwUQe3MxPuau"}, {name: "Rosal√≠a", id:"7ltDVBr6mKbRvohxheJ9h1"}];
// const INITIAL_STATE = {songs:[], artists: SEED_ARTISTS, albums:[]};
import {GET_TRANSLATION} from "./actionTypes";
const ERRORS = {searchError: false, lyricsError: false, translationError: false}
const INITIAL_STATE = {translation: "", errors: ERRORS }

function rootReducer(state=INITIAL_STATE, action) {
    switch(action.type) {
        case GET_TRANSLATION:
        
            // const updatedStateWithErrors = {...state, state.errors.lyricsError = action.errors.lyricsError}
        return {...state, translation: action.translation}

        default:
            return state;
    }
}

export default rootReducer;
