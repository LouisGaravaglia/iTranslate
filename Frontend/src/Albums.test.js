import React from 'react'
import {render, fireEvent, waitFor, screen, cleanup} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect'
import { Provider } from 'react-redux'
import { createStore } from "redux";
import rootReducer from "./reducers/rootReducer";
// import configureStore from 'redux-mock-store'
import Albums from "./Albums";

afterEach(cleanup);

const startingState = {albums: [{albumId: 42}]};

//REDUCER FUNCTION TO MIMIC REDUX REDUCER FOR STORE
function reducer(state = startingState, action) {
  switch (action.type) {
    case "ADD_ALBUMs":
      return {...state, albums: action.albums};
    default:
      return state;
  };
};

//FUNCTION TO ALLOW THE COMPONENT TO BE RENDERED USING OUR MAKESHIF REDUX STORE
function renderWithRedux(
  component,
  {initialState, store = createStore(reducer, initialState)} = {}
) {
  return {
    ...render(<Provider store={store}>{component}</Provider>)
  };
};

//SMOKE TEST
describe('Smoke Test for Albums component', () => {

  it('renders without crashing', () => {
    renderWithRedux(<Albums />)
  });

});

//SNAPSHOT TEST
describe('Snapshot Test for Albums component', () => {

  it('matches snapshot', () => {
    const {asFragment} = renderWithRedux(<Albums />)
    expect(asFragment()).toMatchSnapshot();
  });

});