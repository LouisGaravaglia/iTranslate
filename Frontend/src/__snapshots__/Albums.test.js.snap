// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test for Albums component matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="opacity: 0;"
  >
    <div
      class="Main-Container"
    >
      <div
        class="Browse-Albums"
      >
        <div
          class="Album"
        >
          <img
            alt=""
          />
        </div>
      </div>
      <div
        class="Main-Pagination-Slider-Placeholder"
      />
    </div>
  </div>
</DocumentFragment>
`;
